### PR TITLE
chore(UI): Clean up the UI and remove branded colors

### PR DESCRIPTION
**Pull request by InfamousVague**
### What this PR does 📖

Attempts to remove all "branded" colors on the application. With the use of themes users can pick their favorite colors easily, I think it'd be interesting to remove our opinion on what color people use and really encourage them to make it their own buy just sticking with a grayscale color pallet by default. This also makes it a lot easier contrast wise for most and helps see where we could improve in the UI (things being too dim, or too bright, etc)

image

### CODE REVIEW

**Comment by WanderingHogan** [comment]
clippy angry

**Comment by InfamousVague** [comment]
Changes several colors across the board and increases readability and contrast by default. 

adds ability to click an image to focus it, it will not actually make the picture bigger until we load the real source instead of the thumbnail

adds youtube video link expansion

cleans up multiple images in one message

File: kit/src/components/message/mod.rs
Comment (lgmarchi) [inline kit/src/components/message/mod.rs:91]: Matt, would it not be better to add a doc comment like this: 

- `(?i)`: Enables case-insensitive matching.
- `\b`: Word boundary.
- `(https?://|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)`: Matches the start of a URL. It can start with `http://`, `https://`, `www.`, or any domain name (a sequence of alphanumeric characters, dots, and dashes) followed by a top-level domain (two to four letters).
- `(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+`: Matches the rest of the URL. It can be a sequence of non-space, non-bracket, non-angle bracket characters, or such sequences enclosed in brackets. This can be repeated one or more times.
- `(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’])`: Matches the end of the URL. It can either be a sequence of characters (excluding some punctuation marks) inside brackets, or a single character excluding some specific punctuation marks.

File: kit/src/elements/select/style.scss
Comment (lgmarchi) [inline kit/src/elements/select/style.scss:9]: I think it can be inside our app, because if one day, this website becomes offline, or remove this image, will break this background

**Review by lgmarchi — COMMENTED** [review]
Contrast theme is with wrong text color 
image

Lunar Eclipse is with black text, difficult to view 
image

In default theme, Add friend button, is very difficult to see, even disabled 
image

Check box not showing on default theme 
image

Same thing inside edit group: 
image

**Comment by InfamousVague** [comment]
> Contrast theme is with wrong text color image
> 
> Lunar Eclipse is with black text, difficult to view image
> 
> In default theme, Add friend button, is very difficult to see, even disabled image
> 
> Check box not showing on default theme image
> 
> Same thing inside edit group: image

Don't worry about the contrast theme for now it's still a work in progress. The opacity issues with the disabled buttons is a bug which was introduced because we're over applying the disabled class 2-5 times in some places, causing things to become almost transparent. 

Since there is a lot of work going on in these disabled areas i chose not to resolve them in this PR and think we should do this in a new PR to keep conflicts low. 

Will patch up the checkboxes, though.

File: kit/src/elements/select/style.scss
Comment (InfamousVague) [reply kit/src/elements/select/style.scss:9]: This is all included in the app, the entire svg is just the polygon. If you mean the `xmlns='http://www.w3.org/2000/svg'` I had to look that up because I didn't know what it was actually for (I copy pasted this svg code off stackoverflow.), it doesn't seem to be an actual link to something but instead a required string that exist on a svg. See more info here: https://stackoverflow.com/questions/18467982/are-svg-parameters-such-as-xmlns-and-version-needed

**Comment by lgmarchi** [comment]
> > Contrast theme is with wrong text color image
> > Lunar Eclipse is with black text, difficult to view image
> > In default theme, Add friend button, is very difficult to see, even disabled image
> > Check box not showing on default theme image
> > Same thing inside edit group: image
> 
> Don't worry about the contrast theme for now it's still a work in progress. The opacity issues with the disabled buttons is a bug which was introduced because we're over applying the disabled class 2-5 times in some places, causing things to become almost transparent.
> 
> Since there is a lot of work going on in these disabled areas i chose not to resolve them in this PR and think we should do this in a new PR to keep conflicts low.
> 
> Will patch up the checkboxes, though.

If you don't mind, could you add "Contrast (WIP)", at least, I think it is nice to let it clear

**Review by lgmarchi — APPROVED** [review]

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 29 59

maybe this orange could be red instead

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 30 55

the grey out text seems it isn't clickable but it is

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 31 12

as lucas also mentioned maybe we could add WIP here

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 37 54

in the green theme even thought the edit group is deactivated is not very visible

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 39 20


in the light theme the badges are displayed like this

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 40 22

the icon is behind the download update text in the light theme

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 40 36

in the lunar eclipse the text is too dark

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 43 03


in the polar theme the avatars are too dark

and when hovering the text is too light

Captura de ecrã 2023-06-14, às 14 43 18

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 42 35

it says pink but seems more a bit of orange

**Comment by stavares843** [comment]
Captura de ecrã 2023-06-14, às 14 44 58


in the light theme you can see behind the avatar


doesn't happen in other themes

Captura de ecrã 2023-06-14, às 14 41 51

**Comment by stavares843** [comment]
also i noticed the preview images doesn't work in this PR

Captura de ecrã 2023-06-14, às 14 30 26


works in dev


Captura de ecrã 2023-06-14, às 14 46 25

**Comment by stavares843** [comment]
also the modal is not placed correctly

this PR

image


vs dev



Captura de ecrã 2023-06-14, às 14 47 56

**Review by stavares843 — DISMISSED** [review]
added some feedback in the comments


macOS, m1

**Comment by InfamousVague** [comment]
@stavares843 I went ahead and removed all the themes but the default one. We don't really have time to maintain them all right now and might be best left to the community to create.

Working on the file preview in files page now.

**Comment by stavares843** [comment]
sounds good, thanks! 🎉

**Comment by stavares843** [comment]
feel free to merge this in, we can make tickets if there are any issues 🎉

**Review by WanderingHogan — APPROVED** [review]

**Comment by InfamousVague** [comment]
Before merging, there is one outstanding thing I need to fix, the update modal. I'm not sure how to get it to open up though. Should be good to go after that though. 


I think the removal of themes is better in the long run because none of our themes were really "finished" and having just one theme to focus on for now really simplifies things.

**Pull request merged**
